pkg/providers/amifamily/al2.go

```diff
@@ -33,16 +33,34 @@ type AL2 struct {
 	*Options
 }
 
-// SSMAlias returns the AMI Alias to query SSM
-func (a AL2) SSMAlias(version string, instanceType *cloudprovider.InstanceType) string {
-	amiSuffix := ""
-	if !resources.IsZero(instanceType.Capacity[v1alpha1.ResourceNVIDIAGPU]) || !resources.IsZero(instanceType.Capacity[v1alpha1.ResourceAWSNeuron]) {
-		amiSuffix = "-gpu"
-	}
-	if instanceType.Requirements.Get(v1.LabelArchStable).Has(v1alpha5.ArchitectureArm64) {
-		amiSuffix = fmt.Sprintf("%s-%s", amiSuffix, v1alpha5.ArchitectureArm64)
+// DefaultAMIs returns the AMI name, and Requirements, with an SSM query
+func (a AL2) DefaultAMIs(version string) []DefaultAMIOutput {
```

Comment:
Sanity check: Do we know what happens here when we request something that is using ARM64 and a GPU? I think we should have no compatible AMIs and just return an error with no AMIs that are matching, but just want to double-check

Reply:
Yeah, that is the behavior we see here